activerecord/lib/active_record/connection_adapters/column.rb

```diff
@@ -77,7 +90,7 @@ def hash
       protected
 
         def attributes_for_hash
-          [self.class, name, default, sql_type_metadata, null, table_name, default_function, collation]
+          [self.class, name, name.encoding, default, sql_type_metadata, null, table_name, default_function, collation]
```

Comment:
This made a column encoding test fail. Turns out `String#hash` totally ignores the `String#encoding`.

```ruby
>> "foo".force_encoding('UTF-8').hash == "foo".force_encoding('EUC-JP').hash
=> true
```

Sounds a bit like an MRI bug to me.